### PR TITLE
Book editing and testable examples

### LINKED ISSUE #88: Book: Test code blocks on CI
mdbook has the ability to run rustdoc tests, but I've never set that up. We'll probably need to have several external crates available. Most code examples will also need to the have some imports added to them (but they can be hidden).

### PR DESCRIPTION
**Pull request by killercup**
Adjusting book contents to so people can directly use the code. I might also add some more info on the exmaple code in the repo.

Fixes #88

### CODE REVIEW

**Comment by spacekookie** [comment]
I'll review the rest of it when I'm coughing a bit less :sweat_smile:

File: src/in-depth/config-files.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ and requires you to specify the config layout
 via a `struct` (that is `Serialize`, `Deserialize`)
 and it will figure out the rest!
 
-```rust
+```rust,ignore
```
[the remark this answers is not in the document]
Comment (killercup) [reply src/in-depth/config-files.md]: `mdbook test` would try to compile it, see that `ConfyConfig` isn't defined and fail…

It's not trivial to make code blocks using external deps work, so I replaced most others with actual files in the new cargo project. Haven't gotten around to doing that for the in-depth chapters.

**Comment by killercup** [comment]
Updated. I think this now resolves all open issues with the tutorial except for fixes to the packages chapter.

**Comment by killercup** [comment]
To make it easier to build new chapters on the testing provided here, I'm gonna merge this and to revisions later.

As always, it'll be reviewable here or in #58 :)

**Pull request merged**